interview-questions.md

```diff
@@ -8,62 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: parameters are the template for what goes in the argument. The parameters are asking for a value and you supply the value in the argument. For example, the parameters could be color, size, and flavor. The arguments would be pink, small, and vanilla.
 
-  Researched answer:
+  Researched answer: Parameters are variables that act as placeholders inside the parentheses of the function. Arguments are the things that are passed into the function when it's invoked. We must have the same number of parameters as we do arguments as they correspond.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: arrays but I know there are 2 more I just cannot remember
 
-  Researched answer:
+  Researched answer: value, indexe, array. the index and array are optional whereas values are required.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: map *does something* to each value in an array and returns an array of the same length. filter makes a *decision* about each value in the array and returns and array with the content that meets that filtered criteria.
 
-  Researched answer:
+  Researched answer: My answer was correct for both and I found that map has a return for the outer function as well as a return for higher-order functions. I also found that Filter returns a subset of arrays and, to put it another way, returns only the values that are truthy.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is the action of an object. A method is when a function belongs to an object.
 
-  Researched answer:
+  Researched answer: Functions: reusable blocks of code, live by themselves -is independent, has data passed inside, can be called by it's name, a set of instructions that peform a task, but not directly linked to an object. Methods: are functions linked to an object, uses dot notation to call a method on an object
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer:The re-assignment of variables
 
-  Researched answer:
+  Researched answer: This is the re-assignment of variables so that we don't have to keep repeating ourselves. It keeps our code tighter and cleaner.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Not a clue. Maybe moving something from up one page to another? Did I miss this in class??
```

Comment:
lol, you probably did not miss this.  Stretch questions usually mean that the material may have been somewhere in the reading material but not directly addressed in class.